### PR TITLE
(JP-3338) Change DQ flag and step name for charge migration.

### PR DESCRIPTION
**Pull request by dmggh**
Resolves [JP-3338](https://jira.stsci.edu/browse/JP-3338)

Closes spacetelescope/jwst#7813

 
This PR for JP-3338 modifies the DQ flags and the core schema to support the renaming of the undersampling_correction step to charge_migration.

**Checklist**
- [x] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [ ] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [X] added relevant label(s)

### CODE REVIEW

File: src/stdatamodels/jwst/datamodels/schemas/core.schema.yaml [hunk 1]
```diff
@@ -2212,10 +2212,10 @@ properties:
                 type: string
                 fits_keyword: R_V2V3
                 blend_table: True
-          undersampling_correction:
-            title: Undersampling Correction
+          charge_migration:
```
Comment (hbushouse) [inline src/stdatamodels/jwst/datamodels/schemas/core.schema.yaml:2215]: 1) This whole entry should be moved down in to the "cal_step" section of the header, with all the other "S_xxxxx" keywords. This is the section for reference file names.  2) The title should be "Charge Migration" (mixed case). 3) the keyword name should be "S_CHGMIG", which is more consistent with the step name.

**Review by hbushouse — COMMENTED** [review]
Needs a change log entry.

File: src/stdatamodels/jwst/datamodels/schemas/core.schema.yaml (hunk 1 above)
Comment (dmggh) [reply src/stdatamodels/jwst/datamodels/schemas/core.schema.yaml]: Done

**Review by hbushouse — APPROVED** [review]
Looks good.

**Comment by hbushouse** [comment]
CI failure is due to mismatch in DQ flag name between this PR and master/ramp_fit, which is expected. Will wait to merge this until jwst PR with algorithm updates has been merged.

**Pull request merged**